pkg/build/prometheus/metrics.go

```diff
@@ -0,0 +1,109 @@
+package prometheus
+
+// NOTE, this module explicitly attempts to avoid references to the build api types/structs
+// in an attempt to start breaking up our compile dependencies
+
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// BuildSubsystem is how this build controller is represented in
+// prometheus metrics.
+const BuildSubsystem = "build_controller"
+
+var (
+	// decided not to have a separate counter for failed builds, which have reasons,
+	// vs. the other "finished" builds phases, where the reason is not set
+	buildsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: BuildSubsystem,
+			Name:      "build_counters",
+			Help:      "Counts total teriminal builds by phase, and in case of failures, reason",
+		},
+		[]string{"phase", "reason"},
+	)
+	buildsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: BuildSubsystem,
+			Name:      "build_gauges",
+			Help:      "Show current counts of non-terminal builds by phase",
```

Comment:
ok ... pulled "terminal" from method names in the code, but certainly understand that method names in the code are not *quite* as public facing as this ... of your alternatives, I like active best as well